databases/tiledb/files/patch-linking.diff

```diff
@@ -0,0 +1,47 @@
+--- tiledb/CMakeLists.txt.orig	2023-07-08 03:01:33.000000000 +0800
++++ tiledb/CMakeLists.txt	2023-07-10 17:52:17.000000000 +0800
+@@ -638,6 +638,7 @@
+     ZLIB::ZLIB
+     Zstd::Zstd
+     libmagic
++    MacportsLegacySupport
+ )
+ if (TILEDB_VCPKG)
+   # vcpkg uses SPDLOG_FMT_EXTERNAL=1
+@@ -654,6 +655,13 @@
+   )
+ endif()
+ 
++if(CMAKE_OSX_ARCHITECTURES STREQUAL "ppc")
++  target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
++    INTERFACE
++    atomic
++  )
++endif()
++
+ if(TILEDB_CRC32)
+   find_package(crc32c_EP REQUIRED)
+   find_package(crc32c CONFIG REQUIRED
+
+--- test/CMakeLists.txt.orig	2023-04-26 09:28:54.000000000 +0800
++++ test/CMakeLists.txt	2023-05-22 15:10:49.000000000 +0800
+@@ -295,6 +295,7 @@
+     TILEDB_CORE_OBJECTS_ILIB
+     Catch2::Catch2
+     tiledb_test_support_lib
++    MacportsLegacySupport
+ )
+ 
+ target_link_libraries(tiledb_unit PRIVATE $<BUILD_INTERFACE:common>)
+
+--- cmake/Modules/FindMagic_EP.cmake.orig	2023-04-26 09:28:54.000000000 +0800
++++ cmake/Modules/FindMagic_EP.cmake	2023-05-18 11:33:39.000000000 +0800
+@@ -134,7 +134,7 @@
+         -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+         -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+         -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+-        "-DCMAKE_C_FLAGS=${CFLAGS_DEF}"
++        "-DCMAKE_C_FLAGS=${CFLAGS_DEF} -isystem@PREFIX@/include/LegacySupport -Wl,@PREFIX@/lib/libMacportsLegacySupport.a"
```

Comment:
we don't generally static link this library unless it is something fragile like a compiler. Any subsequent fixes to legacysupport will require knowing about this static link and revbumping this port.

Reply:
@kencu It is generally safe to assume that if I do something this non-standard, there are (or at least were at the time) reasons for that. I am a lazy person, and whenever not, there are more entertaining stuff to do than patching legacy-support linking :)

Admittedly, I have this port building for quite a while, and I did not try to fix legacy-support in a better way with every update. When I did, it failed to work normally. Perhaps because the build system is weird, and builds external components in separate goes.

But if you have a better suggestion, I am eager to try. I do not like the way it had to be done, it is bothersome to carry this patch.

Reply:
there was no reason to link legacysupport statically, AFAICT.

Reply:
Just swap the flag there to dynamic one? I can try that.